### PR TITLE
Update Frame Selection Time to use Epsilon

### PR DESCRIPTION
**Pull request by EricJIsaac**
**Pull request recommendations:**

- [ ] Name your pull request _your-development-type/short-description_. Ex: _feature/read-tiff-files_
- [ ] Link to any relevant issue in the PR description. Ex: _Resolves [gh-##], adds tiff file format support_
- [ ] Provide description and context of changes.
- [ ] Provide relevant tests for your feature or bug fix.
- [ ] Provide or update documentation for any feature added by your pull request.

Thanks for contributing!

Resolves: https://github.com/allen-cell-animated/simularium-engine/issues/14

Instead of doing integer division select a frame, the round function is now used. 
This will select the nearest frame instead of the lower frame.

e.g. with a time-step of 1,
Previously, requesting time 4.9 returned frame 4
Now, requesting time 4.9 will return frame 5

### CODE REVIEW

File: src/main/simulation.cpp [hunk 1]
```diff
@@ -320,14 +320,19 @@ namespace simularium {
...
-            int time = simulationTimeNs / tfp.timeStepSize;
-            return time;
+            // Return the nearest frame based on a fixed time-step size
+            //  e.g. timestep = 2, requestedTime = 5.1,
+            //   round(5.1/2) = round(2.55) = frame 3
+            std::size_t frameNum = std::round(simulationTimeNs / tfp.timeStepSize);
+            return std::min(
+              frameNum,
+              tfp.numberOfFrames - 1
+            );
```
Comment (toloudis) [inline src/main/simulation.cpp:335]: This function is a prime candidate for some unit tests to maintain that all assumptions and edge cases are met. (May possibly need some refactoring to make it more testable)

Comment (EricJIsaac) [reply src/main/simulation.cpp:335]: The unit tests are here:
https://github.com/allen-cell-animated/simularium-engine/blob/main/src/test/test_sim_time.cpp#L37

It was designed from the outset to be testable. The changes here correspond to a change in expected behavior (more specified) - I'll update the tests to reflect the update.

Comment (toloudis) [reply src/main/simulation.cpp:335]: Thanks.  Would those new test cases have failed before the changes? 
I'd feel even more comfortable with some tests of floating point numbers at firstframe +/- epsilon and last frame +/- epsilon too.

Comment (EricJIsaac) [reply src/main/simulation.cpp:335]: Yes, the new test cases would have failed before the changes (the rule was to always round down). Good idea, I'll add those as well.

**Review by toloudis — APPROVED** [review]

**Review by schoinh — APPROVED** [review]
